src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -51,6 +61,23 @@ def get_observation_histograms(self):
         """
         return self._scheduler.get_observation_histograms()
 
+    def get_expanded_specs(self) -> Dict[str, List[str]]:
+        """Return a dict containing the canonicalized addresses of the specs for this run, and what
+        files they expand to."""
+
+        (addresses,) = self._scheduler.product_request(
```

Comment:
redundant parentheses

Reply:
`product_request` returns a list (with one element in this case). We use this kind of destructuring in several places when calling `product_request`

Reply:
a list or a tuple ? I didn't know you can unpack a list this way... I thought this syntax is only for tuples.
@Eric-Arellano wdyt?

Reply:
Asher means that in Python, you can do this instead:

```python
        addresses, = self._scheduler.product_request(
```

Reply:
You can do all sorts of crazy things:
```
$ python
Python 3.9.0 (default, Oct  8 2020, 15:54:02)
[Clang 9.0.8 (https://android.googlesource.com/toolchain/llvm-project 98c855489 on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a, = [42]
>>> a
42
>>> [a,] = [42]
>>> a
42
>>> a, = range(1)
>>> a
0
>>>
```